hikari/impl/stateful_cache.py

```diff
@@ -1598,3 +1602,60 @@ def update_voice_state(
         cached_voice_state = self.get_voice_state(voice_state.guild_id, voice_state.user_id)
         self.set_voice_state(voice_state)
         return cached_voice_state, self.get_voice_state(voice_state.guild_id, voice_state.user_id)
+
+    def clear_messages(self) -> cache.CacheView[snowflakes.Snowflake, messages.Message]:
+        if not self._message_entries:
+            return cache_utility.EmptyCacheView()
+
+        cached_messages = self._message_entries.copy()
+        self._message_entries.clear()
+
+        return cache_utility.StatefulCacheMappingView(cached_messages)
+
+    def delete_message(self, message_id: snowflakes.Snowflake, /) -> typing.Optional[messages.Message]:
+        return self._message_entries.pop(message_id) if message_id in self._message_entries else None
+
+    def get_message(self, message_id: snowflakes.Snowflake, /) -> typing.Optional[messages.Message]:
+        message = self._message_entries.get(message_id)
+        if message:
+            message = copy.copy(message)
+            message.author = copy.copy(message.author)
+
+        return message
+
+    def get_messages_view(self) -> cache.CacheView[snowflakes.Snowflake, messages.Message]:
+        cached_messages = self._message_entries.freeze()
+        return cache_utility.StatefulCacheMappingView(cached_messages)
+
+    def set_message(self, message: messages.Message, /) -> None:
+        self._message_entries[message.id] = copy.copy(message)
+
+    def update_message(
+        self, message: typing.Union[messages.PartialMessage, messages.Message], /
+    ) -> typing.Tuple[typing.Optional[messages.Message], typing.Optional[messages.Message]]:
+        cached_message = self.get_message(message.id)
+        msg = copy.copy(cached_message)
```

Comment:
copied_message might be a nicer name than "msg" for this variable btw and we only have to copy this if message is PartialMessage so might want to copy it in that block of code.